day_10/task_2/pipe_maze.cpp

```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
 {
     return point.first >= 0 and point.first < static_cast<int>(maze_[0].size()) and
         point.second >= 0 and point.second < static_cast<int>(maze_.size());
-}
\ No newline at end of file
+}
+
+MazePoint operator+(const MazePoint& lhs, const MazePoint& rhs)
+{
+    return MazePoint{lhs.first + rhs.first, lhs.second + rhs.second};
+}
+
+void AttributedMaze::color_neighbor(const MazePoint& loop_point, const MazePoint& next_loop_point)
+{
+    assert(is_in_maze(loop_point));
+    assert(is_in_maze(next_loop_point));
+    auto neighbors = find_next_neighbors(loop_point);
+    assert((next_loop_point == neighbors.first || next_loop_point == neighbors.second));
+    MazePoint second_neighbor =
+        (next_loop_point == neighbors.first) ? neighbors.second : neighbors.first;
+
+    std::vector<MazePoint> all_surrounding_tiles = {
+        MazePoint{-1, -1}, MazePoint{-1, 0}, MazePoint{-1, 1},
+        MazePoint{0, 1},   MazePoint{1, 1},  MazePoint{1, 0},
+        MazePoint{1, -1},  MazePoint{0, -1}};
+
+    std::transform(all_surrounding_tiles.begin(), all_surrounding_tiles.end(),
```

Comment:
global comment
consider to extract to a new function in namespace {} and add more descriptive name